pkg/virt-controller/watch/clone/vm-target.go

```diff
@@ -57,18 +62,27 @@ func generatePatches(source *k6tv1.VirtualMachine, cloneSpec *clonev1alpha1.Virt
 	return patches
 }
 
-func generateMacAddressPatches(interfaces []k6tv1.Interface, newMacAddresses map[string]string) (patches []string) {
-	const macAddressPatchPattern = `{"op": "replace", "path": "/spec/template/spec/domain/devices/interfaces/%d/macAddress", "value": "%s"}`
-
+func generateMacAddressPatches(interfaces []k6tv1.Interface, newMacAddresses map[string]string) (patches []string, err error) {
+	// If the newMacAddresses map is nil, no patches would be generated.
+	if newMacAddresses == nil {
+		return
+	}
+	patchSet := patch.New()
 	for idx, iface := range interfaces {
 		// If a new mac address is not specified for the current interface an empty mac address would be assigned.
 		// This is OK for clusters that have Kube Mac Pool enabled. For clusters that don't have KMP it is the users'
 		// responsibility to assign new mac address to every network interface.
 		newMac := newMacAddresses[iface.Name]
-		patches = append(patches, fmt.Sprintf(macAddressPatchPattern, idx, newMac))
+		patchSet.AddOption(
+			patch.WithAdd(fmt.Sprintf("/spec/template/spec/domain/devices/interfaces/%d/macAddress", idx), newMac),
+		)
+	}
+	payload, err := patchSet.GeneratePayload()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate macAddress patch payload: %w", err)
 	}
 
-	return patches
+	return []string{strings.Trim(string(payload), "[]")}, nil
```

Comment:
Why do you still need to trim the string? The idea behind of the PatchSet is to avoid all this string manipulation

Reply:
If the patch isn't complete, you can always return the patchset and add the rest of the patching operation in the other functions

Reply:
@alicefr Moring! yes, I have to deal with this return value. Because the function `func generatePatches()` its return value is `[]string`.

Reply:
I will add this function for `PatchSet`

Reply:
I would change the function signature and simply return []bytes